src/test/clojure/clara/test_rules.clj

```diff
@@ -1183,75 +1183,85 @@
                                  [?t <- Temperature (= ?l location)]
                                  [:not [Cold (= temperature (:temperature ?t))]]]]])
 
-        s (mk-session [cold-not-match-temp] :cache false)
-        s-with-prior (mk-session [negation-with-prior-bindings] :cache false)
-        s-with-nested (mk-session [nested-negation-with-prior-bindings] :cache false)]
+        object-query (dsl/parse-query [] [[?o <- Object]])
 
-    (is (= [{}]
-           (-> s
-               (fire-rules)
-               (query cold-not-match-temp))))
+        s (mk-session [cold-not-match-temp object-query] :cache false)
+        s-with-prior (mk-session [negation-with-prior-bindings object-query] :cache false)
+        s-with-nested (mk-session [nested-negation-with-prior-bindings object-query] :cache false)
+
+        no-system-types? (fn [session]
+                           (not-any? (fn [fact] (isa? (type fact) :clara.rules.engine/system-type))
+                                     (as-> session r
```

Comment:
Ah yes.  I see how `as->` can get awkward since the binding really represents multiple phases in a thread of results.  That's fair.  I was thinking `r` for `session` was weird, but I see how you use it later as the query result.

I don't know what the idiom is for this one.  Maybe just `x` or something that doesn't try to name anything specifically.  It doesn't matter though.  I asked more out of curiosity here.